lib/chef/resource/kernel_module.rb

```diff
@@ -0,0 +1,128 @@
+#
+# Resource:: kernel_module
+#
+# The MIT License (MIT)
+#
+# Copyright 2016-2018, Shopify Inc.
+# Copyright 2018, Chef Software, Inc.
+
+require "chef/resource"
+
+class Chef
+  class Resource
+    class KernelModule < Chef::Resource
+      preview_resource true
+      resource_name :kernel_module
+
+      description "Use the kernel_module resource to manage kernel modules on Linux systems. This resource can load, unload, blacklist, install, and uninstall modules."
+      introduced "14.3"
+
+      property :modname, String,
+               description: "The name of the kernel module.",
+               name_property: true, identity: true
+
+      property :load_dir, String,
+               description: "The directory to load modules from.",
+               default: "/etc/modules-load.d"
+
+      property :unload_dir, String,
+               description: "The modprobe.d directory.",
+               default: "/etc/modprobe.d"
+
+      action :install do
+        description "Load kernel module, and ensure it loads on reboot"
+
+        declare_resource(:directory, new_resource.load_dir) do
+          recursive true
+        end
```

Comment:
its custom-resource style so i don't think you need to do `declare_resource`?

Reply:
Don't need to, but doesn't hurt anything so authors style choice.

Reply:
Fixed